### PR TITLE
feat: Added P.string.length

### LINKED ISSUE #256: P.string.length
**Is your feature request related to a problem? Please describe.**
A subpattern that can be composed with P.intersect, or P.union. .when(x=>x.length) could handle this, but it doesn't guarantee string type.
the other alternative, P.intersection(P.string.minLength(len), P.string.maxLength(len)) could be equivalent, and also can be done with APIs we already have, but it's very cumbersome and the boundaries of those min and maxes(includes or not) are confusing. 

**Describe the solution you'd like**
Add P.string.length(len: number), which matches the given value's type is string and its length is same as given `len`

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by JUSTIVE**
resolves #256 .

### CODE REVIEW

File: src/types/Pattern.ts [hunk 1]
```diff
@@ -313,6 +313,21 @@ export type StringChainable<
         MergeGuards<input, p, GuardExcludeP<unknown, string, never>>,
         omitted | 'minLength'
       >;
+      /**
+       * `P.string.length(len)` is a pattern, matching **strings** with exact `len` characters.
```
Comment (gvergnaud) [inline src/types/Pattern.ts:317]: did you mean
```suggestion
       * `P.string.length(len)` is a pattern, matching **strings** with exactly `len` characters.
```
?

**Review by gvergnaud — COMMENTED** [review]
Thanks for your PR! a few minor comments

**Comment by JUSTIVE** [comment]
thanks for reply! applied all of your comments

**Comment by gvergnaud** [comment]
Merged and published as 5.2.0

**Pull request merged**